precompiles/xcm-transactor/XcmTransactor.sol

```diff
@@ -30,55 +32,91 @@ interface XcmTransactor {
+        external
+        view
+        returns (
+            uint64 transact_extra_weight,
+            uint64 transact_extra_weight_signed,
+            uint64 max_weight
+        );
 
     /** Get fee per second charged in its reserve chain for an asset
      * Selector 83f09082
      * @param multilocation The asset location for which we want to know the fee per second value
      * @return fee_per_second The fee per second that the reserve chain charges for this asset
      */
-    function fee_per_second(Multilocation memory multilocation) external view 
-        returns(uint256 fee_per_second);
+    function fee_per_second(Multilocation memory multilocation)
+        external
+        view
+        returns (uint256 fee_per_second);
 
     /** Transact through XCM using fee based on its multilocation
-    * Selector 9f89f03e
-    * @dev The token transfer burns/transfers the corresponding amount before sending
-    * @param transactor The transactor to be used
-    * @param index The index to be used
-    * @param fee_asset The asset in which we want to pay fees. 
-    * It has to be a reserve of the destination chain
-    * @param weight The weight we want to buy in the destination chain
-    * @param inner_call The inner call to be executed in the destination chain
-    */
+     * Selector 9f89f03e
+     * @dev The token transfer burns/transfers the corresponding amount before sending
+     * @param transactor The transactor to be used
+     * @param index The index to be used
+     * @param fee_asset The asset in which we want to pay fees.
+     * It has to be a reserve of the destination chain
+     * @param weight The weight we want to buy in the destination chain
+     * @param inner_call The inner call to be executed in the destination chain
+     */
     function transact_through_derivative_multilocation(
         uint8 transactor,
         uint16 index,
         Multilocation memory fee_asset,
         uint64 weight,
         bytes memory inner_call
     ) external;
-    
+
+    /** Transact through XCM using fee based on its multilocation
+     * Selector 9f89f03e
+     * @dev The token transfer burns/transfers the corresponding amount before sending
+     * @param transactor The transactor to be used
+     * @param index The index to be used
+     * @param fee_asset The asset in which we want to pay fees.
+     * It has to be a reserve of the destination chain
+     * @param transact_require_weight_at_most The weight we want to buy in the destination chain
+     * @param inner_call The inner call to be executed in the destination chain
+     * @param fee_amount Amount to be used as fee.
+     * @param overall_weight Overall weight to be used for the xcm message.
+     */
+    function transact_through_derivative_multilocation(
+        uint8 transactor,
+        uint16 index,
+        Multilocation memory fee_asset,
+        uint64 transact_require_weight_at_most,
```

Comment:
should `transact_require_weight_at_most` be `transact_required_weight_at_most` ? (not sure)

Reply:
require_weight_at_most is the name of the field in the Transact instruction. But I think your suggestion better expresses the intention of the param

Reply:
changed